File: SDKs/googleapis.com/vision/v1p1beta1/go/pkg/models/shared/googlecloudvisionv1p1beta1colorinfo.go
package shared

// GoogleCloudVisionV1p1beta1ColorInfo
// Color information consists of RGB channels, score, and the fraction of the image that the color occupies in the image.
type GoogleCloudVisionV1p1beta1ColorInfo struct {
	Color         *Color   `json:"color,omitempty"`
	PixelFraction *float32 `json:"pixelFraction,omitempty"`
	Score         *float32 `json:"score,omitempty"`
}

File: SDKs/googleapis.com/vision/v1p1beta1/typescript/src/sdk/models/shared/googlecloudvisionv1p1beta1colorinfo.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import { Color } from "./color";


// GoogleCloudVisionV1p1beta1ColorInfo
/** 
 * Color information consists of RGB channels, score, and the fraction of the image that the color occupies in the image.
**/
export class GoogleCloudVisionV1p1beta1ColorInfo extends SpeakeasyBase {
  @Metadata({ data: "json, name=color" })
  color?: Color;

  @Metadata({ data: "json, name=pixelFraction" })
  pixelFraction?: number;

  @Metadata({ data: "json, name=score" })
  score?: number;
}
